include/nuttx/arch.h

```diff
@@ -2278,7 +2253,8 @@ void weak_function nxsched_process_cpuload(void);
  ****************************************************************************/
 
 #if defined(CONFIG_SCHED_CPULOAD) && defined(CONFIG_SCHED_CPULOAD_EXTCLK)
-void weak_function nxsched_process_cpuload_ticks(uint32_t ticks);
+void nxsched_process_cpuload_ticks(uint32_t ticks);
+#  define nxsched_process_cpuload() nxsched_process_cpuload_ticks(1)
```

Comment:
maybe define part can be moved to `sched/sched/sched.h`?

Reply:
No, since the chip need call nxsched_process_cpuload_ticks when CONFIG_SCHED_CPULOAD_EXTCLK is defined.

Reply:
Still not clear to me why we can't
```
#if !defined(CONFIG_SCHED_CPULOAD_EXTCLK)

/* CPU load measurement support */

#  ifdef CONFIG_SCHED_CPULOAD
void nxsched_process_cpuload_ticks(uint32_t ticks);
#  else
#    define nxsched_process_cpuload_ticks(ticks)
#  endif

#  define nxsched_process_cpuload() nxsched_process_cpuload_ticks(1)
#elif defined(CONFIG_SCHED_CPULOAD)
# define nxsched_process_cpuload() nxsched_process_cpuload_ticks(1)
#endif
```

Reply:
Or even
```
/* CPU load measurement support */

#ifdef CONFIG_SCHED_CPULOAD
void nxsched_process_cpuload_ticks(uint32_t ticks);
#else
#  define nxsched_process_cpuload_ticks(ticks)
#endif

#define nxsched_process_cpuload() nxsched_process_cpuload_ticks(1)
```

I mean that in `include/nuttx/arch.h` the `# define nxsched_process_cpuload() nxsched_process_cpuload_ticks(1)` in case if `CONFIG_SCHED_CPULOAD_EXTCLK` is defined and in `sched/sched/sched.h` the `# define nxsched_process_cpuload() nxsched_process_cpuload_ticks(1)` if `CONFIG_SCHED_CPULOAD_EXTCLK` is not defined, so this seems to be duplication for me. Am I missing something?

Reply:
In case if CONFIG_SCHED_CPULOAD_EXTCLK is defined, chip call can call either nxsched_process_cpuload or nxsched_process_cpuload_ticks. If we only define nxsched_process_cpuload in sched/sched/sched.h, how the code in arch/src/xxx call it?